src/metadata.rs

```diff
@@ -165,6 +171,28 @@ impl Metadata {
         }
         string
     }
+
+    /// Derive a mortal period
+    pub(crate) fn derive_mortal_period(&self) -> Result<u64, MetadataError> {
+        let block_hash_count: u64 = self
+            .module("System")
+            .and_then(|sys| sys.constant("BlockHashCount"))
+            .and_then(|count| count.value::<u32>())
+            .map(Into::into)?;
+        let expected_block_time = self
+            .module("Babe")
+            .and_then(|babe| babe.constant("ExpectedBlockTime"))
```

Comment:
Is it fair to assume that `BABE` is used for all runtimes?

Reply:
No, I think Aura is popular as well.

Reply:
I think we can use TimeStamp::MinimumPeriod and double it

https://github.com/paritytech/substrate/blob/332399d16668a6c769f1a7db154bb9ea3b50e61c/frame/timestamp/src/lib.rs#L129-L133